packages/@react-aria/interactions/src/useFocusVisible.ts

```diff
@@ -210,15 +228,17 @@ const tearDownWindowFocusTracking = (element, loadListener?: () => void) => {
  * @returns A function to remove the event listeners and cleanup the state.
  */
 export function addWindowFocusTracking(element?: HTMLElement | null): () => void {
-  const documentObject = getOwnerDocument(element);
+  const rootNode = getRootNode(element);
   let loadListener;
-  if (documentObject.readyState !== 'loading') {
+
+   // Shadow root doesn't have a readyState, so we can assume it's ready in case of there is a shadow root.
+  if (rootNode instanceof ShadowRoot || (rootNode?.readyState !== 'loading')) {
```

Comment:
will this have a problem with `instanceof`?

Reply:
Yes, it can be problematic, and it is better to update it with `nodeType`; I shall do the update.

Reply:
Update in the last commit.